src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/DelayLoadHelperThunk.cs

```diff
@@ -38,64 +38,48 @@ protected override string GetName(NodeFactory factory)
             return "DelayLoadHelper";
         }
 
-        public override ObjectData GetData(NodeFactory factory, bool relocsOnly)
+        protected override void EmitCode(NodeFactory factory, ref X64.X64Emitter instructionEncoder, bool relocsOnly)
         {
-            ObjectDataBuilder builder = new ObjectDataBuilder();
-            builder.AddSymbol(this);
-
-            switch (factory.Target.Architecture)
+            // lea rax, [pCell]
+            instructionEncoder.EmitLEAQ(X64.Register.RAX, _instanceCell);
+            if (!relocsOnly)
             {
-                case TargetArchitecture.X64:
-                    {
-                        if (!relocsOnly)
-                        {
-                            builder.RequireInitialAlignment(8);
-
-                            // lea rax, [pCell]
-                            builder.EmitByte(0x48);
-                            builder.EmitByte(0x8D);
-                            builder.EmitByte(0x05);
-                        }
-                        builder.EmitReloc(_instanceCell, RelocType.IMAGE_REL_BASED_REL32);
-
-                        if (!relocsOnly)
-                        {
-                            // push table index
-                            builder.EmitByte(0x6A);
-                            builder.EmitByte((byte)_instanceCell.Table.IndexFromBeginningOfArray);
-
-                            // push [module]
-                            builder.EmitByte(0xFF);
-                            builder.EmitByte(0x35);
-                        }
-                        builder.EmitReloc(_moduleImport, RelocType.IMAGE_REL_BASED_REL32);
+                // push table index
+                instructionEncoder.Builder.EmitByte(0x6A);
+                instructionEncoder.Builder.EmitByte((byte)_instanceCell.Table.IndexFromBeginningOfArray);
 
-                        if (!relocsOnly)
-                        {
-                            // TODO: additional tricks regarding UNIX AMD64 ABI
-
-                            // jmp [helper]
-                            builder.EmitByte(0xFF);
-                            builder.EmitByte(0x25);
-                        }
-                        builder.EmitReloc(_helperCell, RelocType.IMAGE_REL_BASED_REL32);
-
-                        break;
-                    }
+                // push [module]
+                instructionEncoder.Builder.EmitByte(0xFF);
+                instructionEncoder.Builder.EmitByte(0x35);
+            }
+            instructionEncoder.Builder.EmitReloc(_moduleImport, RelocType.IMAGE_REL_BASED_REL32);
+            
+            if (!relocsOnly)
+            {
+                // TODO: additional tricks regarding UNIX AMD64 ABI
 
-                default:
-                    throw new NotImplementedException();
+                // jmp [helper]
+                instructionEncoder.Builder.EmitByte(0xFF);
```

Comment:
Why not `instructionEncoder.EmitJmp(helperCell)`? That way the reader doesn't have to trust comments or hand-disassemble the bytes.